torch/nn/modules/container.py

```diff
@@ -62,6 +60,9 @@ def __getitem__(self, idx):
                 next(it)
             return next(it)
 
+    def __setitem__(self, idx, module):
+        return setattr(self, str(idx), module)
```

Comment:
It's not guaranteed that `i`th module will be called `str(i)` - see the constructor that takes an OrderedDict

Reply:
Actually I dislike the `[]` access pattern because it's not consistency with normal `Module`. But here we can't access the `sub modules` in  container because the attribute name is a pure number `0`, `1` ..., how about naming the sub modules with pre-fix such as `sub_module_0`, `sub_module_1`.

Reply:
I'm not sure if I follow. We allow users to specify their own names, and we should keep this contract, so no for renaming the modules. If you want to get `i`th module, you need to iterate over `_modules.values()` until you reach the `i`th value.

Reply:
Iterating the `values()` won't let me to modify the value at all. Instead I work it around by iterating the module name and set the corresponding value.